src/pages/resultsView/fusion/ResultViewFusionMapperStore.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2018 The Hyve B.V.
+ * This code is licensed under the GNU Affero General Public License (AGPL),
+ * version 3, or (at your option) any later version.
+ *
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+import {
+    CancerStudy,
+    Gene,
+    MolecularProfile,
+    SampleIdentifier,
+    StructuralVariant
+} from '../../../shared/api/generated/CBioPortalAPI';
+import { cached, labelMobxPromises, MobxPromise } from "mobxpromise";
+import { computed } from 'mobx';
+import {remoteData} from "public-lib/api/remoteData";
+import { fetchEnsemblTranscriptsByEnsemblFilter, fetchPfamDomainData } from '../../../shared/lib/StoreUtils';
+import { PfamDomain } from '../../../shared/api/generated/GenomeNexusAPI';
+import * as _ from 'lodash';
+import { EnsemblTranscriptExt, StructuralVariantExt } from '../../../shared/model/Fusion';
+import ResultViewFusionMapperDataStore from './ResultViewFusionMapperDataStore';
+import { ExonsChartStore } from '../../../shared/components/exonsCharts/ExonsChartStore';
+
+export const FUS_GENE_COLORSCALE = ['#097AFF', '#3B9CFF', '#28a9ff', '#84B7DF', '#92C8E8', '#68ACEA', '#4eb7d9'];
+export const REF_GENE_COLORSCALE = ['#88C542', '#99c556', '#a0cf31', '#7ba237', '#47cb4a', '#008828', '#005623'];
+
+export class ResultViewFusionMapperStore {
+
+    constructor(public gene: Gene,
+                public studyIdToStudy: MobxPromise<{ [studyId: string]: CancerStudy }>,
+                public molecularProfileIdToMolecularProfile: MobxPromise<{ [molecularProfileId: string]: MolecularProfile }>,
+                public samples: MobxPromise<SampleIdentifier[]>,
+                public fusions: StructuralVariantExt[]) {
+        labelMobxPromises(this);
+    }
+
+    private applyColor (result: EnsemblTranscriptExt[]) {
+        let fusionTranscripts: EnsemblTranscriptExt[] = [];
```

Comment:
this could really be a util since it never uses `this`, should probably be unit tested as well. I still think `applyColor` is not a good name, `apply` is not the right verb. `addColor` works better.